src/main/java/com/sahabuddin/contactmanager/controllers/HomeController.java

```diff
@@ -22,6 +33,36 @@ public String about(Model model) {
     @GetMapping(value = "/signup")
     public String signup(Model model) {
         model.addAttribute("title", "Sign up | Contact Manager");
+        model.addAttribute("user", new User());
+        return "signup";
+    }
+
+    @PostMapping(value = "/do-register")
+    public String userSignup(@ModelAttribute("user") User user, @RequestParam(value = "agreement", defaultValue = "false") boolean agreement, Model model, HttpSession session) {
+        try{
+            if (!agreement) {
+                log.info("Agreement not available");
+            }
+            model.addAttribute("title", "Sign up | Contact Manager");
+            user.setRole("ROLE_USER");
+            user.setEnabled(true);
+            user.setImageUrl("default.png");
+            User saved = userRepository.save(user);
```

Comment:
Write a function in UserService to register user. Call this function from this controller.